deps/rabbitmq_peer_discovery_etcd/src/rabbitmq_peer_discovery_etcd_v3_client.erl

```diff
@@ -230,16 +230,13 @@ connected({call, From}, list_keys, Data = #statem_data{connection_name = Conn})
     rabbit_log:debug("etcd peer discovery: will use prefix ~ts to query for node keys", [Prefix]),
     {ok, #{kvs := Result}} = eetcd_kv:get(C2),
     rabbit_log:debug("etcd peer discovery returned keys: ~tp", [Result]),
-    Values = [maps:get(value, M) || M <- Result],
-    rabbit_log:debug("etcd peer discovery: listing node keys returned ~b results", [length(Values)]),
-    ParsedNodes = lists:map(fun extract_node/1, Values),
-    {Successes, Failures} = lists:partition(fun filter_node/1, ParsedNodes),
-    JoinedString = lists:join(",", [rabbit_data_coercion:to_list(Node) || Node <- lists:usort(Successes)]),
-    rabbit_log:error("etcd peer discovery: successfully extracted nodes: ~ts", [JoinedString]),
-    lists:foreach(fun(Val) ->
-        rabbit_log:error("etcd peer discovery: failed to extract node name from etcd value ~tp", [Val])
-    end, Failures),
-    gen_statem:reply(From, lists:usort(Successes)),
+    Values = [{maps:get(create_revision, M), maps:get(value, M)} || M <- Result],
+    rabbit_log:debug("etcd peer discovery: listing node keys returned ~b results",
+                     [length(Values)]),
+    ParsedNodes = lists:filtermap(fun extract_node/1, Values),
+    rabbit_log:info("etcd peer discovery: successfully extracted nodes: ~0tp",
+                    [ParsedNodes]),
+    gen_statem:reply(From, lists:usort(ParsedNodes)),
```

Comment:
Could you please describe in a comment and the commit message why we sort the nodes and pick the first?